controllers/userController.js

```diff
@@ -194,6 +203,50 @@ exports.getAllUsers = catchAsync(async (req, res, next) => {
   });
 });
 
+exports.deactivateUser = catchAsync(async (req, res, next) => {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(statusCodes.BAD_REQUEST).json({ errors: errors.array() });
+  }
+  const email = req.body.email;
+  const user = await User.findOne({ email });
+  if (!user) {
+    return next(new AppError("User not found", statusCodes.BAD_REQUEST));
+  }
+  if (!user.isActive) {
+    return next(
+      new AppError("User already deactivated", statusCodes.BAD_REQUEST)
```

Comment:
Everything is not always bad request. Always use http code that's relatable to the issue. This can be `CONFLICT` rather than `BAD_REQUEST`